File: android/src/main/kotlin/io/openim/flutter_openim_sdk_ffi/models/VideoElem.kt
package io.openim.flutter_openim_sdk_ffi.models

class VideoElem {
    /**
     * 视频本地资源地址
     */
    var videoPath: String? = null

    /**
     * 视频唯一ID
     */
    var videoUUID: String? = null

    /**
     * 视频oss地址
     */
    var videoUrl: String? = null

    /**
     * 视频类型
     */
    var videoType: String? = null

    /**
     * 视频大小
     */
    var videoSize: Long = 0

    /**
     * 视频时长
     */
    var duration: Long = 0

    /**
     * 视频快照本地地址
     */
    var snapshotPath: String? = null

    /**
     * 视频快照唯一ID
     */
    var snapshotUUID: String? = null

    /**
     * 视频快照大小
     */
    var snapshotSize: Long = 0

    /**
     * 视频快照oss地址
     */
    var snapshotUrl: String? = null

    /**
     * 视频快照宽度
     */
    var snapshotWidth = 0

    /**
     * 视频快照高度
     */
    var snapshotHeight = 0
}

File: ios/Classes/models/VideoElem.swift


public class VideoElem {
    /**
     * 视频本地资源地址
     */
    var videoPath: String? 

    /**
     * 视频唯一ID
     */
    var videoUUID: String? 

    /**
     * 视频oss地址
     */
    var videoUrl: String? 

    /**
     * 视频类型
     */
    var videoType: String? 

    /**
     * 视频大小
     */
    var videoSize: Int64 = 0

    /**
     * 视频时长
     */
    var duration: Int64 = 0

    /**
     * 视频快照本地地址
     */
    var snapshotPath: String? 

    /**
     * 视频快照唯一ID
     */
    var snapshotUUID: String? 

    /**
     * 视频快照大小
     */
    var snapshotSize: Int64 = 0

    /**
     * 视频快照oss地址
     */
    var snapshotUrl: String? 

    /**
     * 视频快照宽度
     */
    var snapshotWidth = 0

    /**
     * 视频快照高度
     */
    var snapshotHeight = 0
}
